ext/standaloneusers/ang/crmChangePassword.js

```diff
@@ -0,0 +1,112 @@
+      hibp: '@',
+      userId: '@'
+    },
+    controller: function($scope, $timeout, crmApi4) {
+      var ts = $scope.ts = CRM.ts(null),
+        ctrl = this;
+
+      console.log('init crmChangePassword component starting');
+      // $onInit gets run after the this controller is called, and after the bindings have been applied.
+      // this.$onInit = function() { console.log('user', ctrl.userId); };
+      ctrl.actorPassword = '';
+      ctrl.newPassword = '';
+      ctrl.newPasswordAgain = '';
+      ctrl.busy = '';
+      ctrl.pwnd = false;
+
+      let updateAngular = (prop, newVal) => {
+        $timeout(() => {
+          console.log("Setting", prop, "to", newVal);
+          ctrl[prop] = newVal;
+        }, 0);
+      };
+
+      ctrl.attemptChange = async () => {
+        updateAngular('busy', '');
+        updateAngular('pwnd', false);
+        updateAngular('formInactive', true);
+        if (ctrl.newPassword.length < 8) {
+          alert(ts("C'mon now, you can do better than that."));
+          return;
+        }
+        if (ctrl.newPassword != ctrl.newPasswordAgain) {
+          alert(ts("Passwords do not match"));
+          return;
+        }
+
+        if (ctrl.hibp) {
+          updateAngular('busy', ts('Checking password is not known to have been involved in data breach...'));
+          let hash = await sha1(ctrl.newPassword)
+          if (!hash.match(/^[a-f0-9]+$/)) {
+            updateAngular('busy', ts('Could not check. Is your browser up-to-date?'));
+          }
+          else {
+            hash = hash.toUpperCase();
+            hashPrefix = hash.substring(0, 5);
+            await fetch(ctrl.hibp + hashPrefix)
+              .then(r => r.text())
+              .then(hibpResult => {
+                if (hibpResult &&
+                  hibpResult.split(/\r\n/).find(line => hashPrefix + line.replace(/:\d+$/, '') === hash)) {
+                  // e.g. Password123
+                  updateAngular('pwn', true);
+                  return;
+                }
+                updateAngular('busy', '');
+              })
+              .catch( () => {
+                updateAngular('busy', ts('Could not perform check; service error. '));
+              });
+          }
+        }
+
+        updateAngular('busy', ctrl.busy + ts('Changing...'));
+
+        // Now submit api request.
+        const userUpdateParams = {
+          hashPassword: 'hash_password',
+          actorPassword: ctrl.actorPassword,
+          values: {password: ctrl.newPassword},
+          where: [['id', '=', ctrl.userId]]
+        };
+        crmApi4('User', 'Update', userUpdateParams)
+        .then(r => {
+            updateAngular('busy', ts('Password successfully updated'));
+          })
+        .catch(e => {
+            let msg = (e.error_message === 'Authorization failed')
+              ? ts('Sorry, that didn\'t work. Perhaps you mistyped your password?')
+              : e.error_message;
```

Comment:
I think the linter is complaining about this being on multiple lines (or expects it in another way)
While you're here, could you use double-quotes to avoid escaping single quotes? I'm not sure how the string extraction handles that (for translation)